src/repo/iRepository.ts

```diff
@@ -28,73 +27,107 @@ interface IRepo<Task>{
+            return getATasks
+        } catch (err) { 
             return null
         }
     }
 
     async getCompleted(val) {
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('SELECT * FROM tasks WHERE completed = ?',{
-                replacements: [val]
-            })
-            return results
-        } catch(err){
+        if (val == 0 || val == 1) {
+            const taskModel = taskModelLink.taskModel();
+            const getCompletedTasks = await taskModel.findAll({
+                where: {
+                    completed: val
+                }
+            });
+            return getCompletedTasks
+        } else {
             return null
         }
     }
 
-    //cant use ? as the replacement value adds quotes which cause order by to ignore specified val
     async sortBy(val) {
-        try{
-            const sortQuery = 'SELECT * FROM tasks ORDER BY ' + val + ' ASC'
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query(sortQuery)
-            return results
-        }catch(err){
+        try {
+            const taskModel = taskModelLink.taskModel();
+            const getSortedTasks = await taskModel.findAll({
+                order: [
+                    [val, 'ASC']
+                ]
+            });
+            return getSortedTasks
+        } catch (err) {
             return null
         }
     }
 
     async updateById(new_id, updated) {
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('UPDATE tasks SET title = ?, dueDate = ?, completed = ? WHERE id = ?',{
-                replacements: [updated.title, updated.dueDate, updated.completed, new_id]
+        try {
+            const taskModel = taskModelLink.taskModel();
+            await taskModel.update({
+                title: updated.title,
+                dueDate: updated.dueDate,
+                completed: updated.completed },
+                {
+                where: {
+                    id: new_id
+                }
+            });
+            //need to return the task that has been updated
+            const updatedTask = await taskModel.findAll({
+                where: {
+                    id: new_id
+                }
             })
-            return results
-        } catch(err){
+            return updatedTask
+        } catch (err) {
             return null
         }
     }
 
     async removeById(new_id) {
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('DELETE FROM tasks WHERE id = ?',{
-                replacements: [new_id]
-            })
-            return results
-        } catch(err){
+        try {
+            const taskModel = taskModelLink.taskModel();
+            const deleteTasks = await taskModel.destroy({
+                where: {
+                    id: new_id
+                }
+            });
+            return deleteTasks
```

Comment:
.destroy just returns how many rows in your table have been deleted; instead, think about what you want returned to your route once the item of id of new_id is deleted